CMakeLists.txt

```diff
@@ -1,22 +1,48 @@
 cmake_minimum_required(VERSION 3.15)
-project(bufr VERSION 11.3.0)
+
+file(STRINGS "VERSION" pVersion)
+
+project(
+    bufr 
+    VERSION ${pVersion}
+    LANGUAGES C Fortran)
+
 set(${PROJECT_NAME}_VERSION ${PROJECT_VERSION} CACHE INTERNAL "${PROJECT_NAME} version number")
-enable_language (Fortran)
 
-if (NOT CMAKE_BUILD_TYPE)
-  set (CMAKE_BUILD_TYPE RELEASE CACHE STRING
-    "Choose the type of build, options are: PRODUCTION Debug Release."
-    FORCE)
+if(NOT CMAKE_BUILD_TYPE MATCHES "^(Debug|Release|RelWithDebInfo|MinSizeRel)$")
+  message(STATUS "Setting build type to 'Release' as none was specified.")
+  set(CMAKE_BUILD_TYPE
+      "Release"
+      CACHE STRING "Choose the type of build." FORCE)
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release"
+                                               "MinSizeRel" "RelWithDebInfo")
+endif()
+
+if(NOT CMAKE_C_COMPILER_ID MATCHES "^(Intel|GNU|Clang|AppleClang)$")
+  message(WARNING "Compiler not officially supported: ${CMAKE_C_COMPILER_ID}")
+endif()
+
+add_compile_definitions(UNDERSCORE)
+if(APPLE)
+  add_compile_definitions(APPLE)
+elseif(UNIX)
+  add_compile_definitions(LINUX)
 endif()
 
-if(CMAKE_CXX_COMPILER_ID STREQUAL "Intel")
+if(CMAKE_C_COMPILER_ID MATCHES "^(Intel)$")
+  include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/Intel.cmake)
   set(IntelComp true )
-elseif(CMAKE_CXX_COMPILER_ID MATCHES "GNU*" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang*")
+elseif(CMAKE_C_COMPILER_ID MATCHES "^(GNU|Clang|AppleClang)$")
   set(GNUComp true )
-elseif(CMAKE_CXX_COMPILER_ID MATCHES "pgc*")
-  set(PGIComp true )
+  include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/GNU.cmake)
 endif()
 
+set(fortran_8_flags ${FREAL8} ${FINT8})
+set(shared_flags  ${DEBINFO} ${FOPT3} ${TRACEBACK} ${AVX2})
```

Comment:
WIth GNU compiler I see these warnings;
```
Scanning dependencies of target bufr_8_c
[ 33%] Building C object CMakeFiles/bufr_8_c.dir/src/arallocc.c.o
cc1: warning: command line option ‘-fbacktrace’ is valid for Fortran but not for C
[ 33%] Building C object CMakeFiles/bufr_8_c.dir/src/ardllocc.c.o
cc1: warning: command line option ‘-fbacktrace’ is valid for Fortran but not for C
[ 33%] Building C object CMakeFiles/bufr_8_c.dir/src/bort_exit.c.o
cc1: warning: command line option ‘-fbacktrace’ is valid for Fortran but not for C
```
``${TRACEBACK}`` should be used in ``shared_flags``